packages/amplify-cli/src/initialize-env.ts

```diff
@@ -42,31 +42,44 @@ export const initializeEnv = async (
     const categoryPluginInfoList = context.amplify.getAllCategoryPluginInfo(context);
     const availableCategories = Object.keys(categoryPluginInfoList).filter(key => initializedCategories.includes(key));
 
-    availableCategories.forEach(category => {
-      categoryPluginInfoList[category].forEach(pluginInfo => {
-        try {
-          // eslint-disable-next-line import/no-dynamic-require, global-require, @typescript-eslint/no-var-requires
-          const { initEnv } = require(pluginInfo.packageLocation);
-
-          if (initEnv) {
-            categoryInitializationTasks.push(() => initEnv(context));
-          }
-        } catch (e) {
-          context.print.warning(`Could not load initEnv for ${category}`);
+    const importCategoryPluginAndQueueInitEnvTask = async (pluginInfo, category) : Promise<void> => {
+      try {
+        const { initEnv } = await import(pluginInfo.packageLocation);
+
+        if (initEnv) {
+          categoryInitializationTasks.push(() => initEnv(context));
         }
-      });
-    });
+      } catch (e) {
+        throw new AmplifyFault('PluginNotLoadedFault', {
+          message: `Could not load plugin for category ${category}. Error: ${e.message}`,
+          resolution: `Review the error message and stack trace for additional information.`,
```

Comment:
you probably want to use `amplifyFaultWithTroubleshootingLink`

Reply:
It is an error to not throw; imaging we are failing silently if a plugin is not imported correctly; we should not proceed to initEnv on the other plugins